internal/pkg/cli/pipeline_init.go

```diff
@@ -295,22 +342,57 @@ func (o *initPipelineOpts) parseOwnerRepoName(url string) (string, string, error
 	parsedURL = strings.TrimSuffix(parsedURL, ".git")
 	ownerRepo := strings.Split(parsedURL, "/")
 	if len(ownerRepo) != 2 {
-		return "", "", fmt.Errorf("unable to parse the GitHub repository owner and name from %s: please pass the repository URL with the format `--github-url https://github.com/{owner}/{repositoryName}`", url)
+		return "", "", fmt.Errorf("unable to parse the GitHub repository owner and name from %s: please pass the repository URL with the format `--url https://github.com/{owner}/{repositoryName}`", url)
 	}
 	return ownerRepo[0], ownerRepo[1], nil
 }
 
+// parseRepoName splits the url on slashes and returns the last substring.
+func (o *initPipelineOpts) parseRepoName(url string) (string, error) {
+	parsedForRepo := strings.Split(url, "/")
+	if len(parsedForRepo) < 2 {
+		return "", fmt.Errorf("unable to parse the CodeCommit repository name from %s", url)
+	}
+	return parsedForRepo[len(parsedForRepo)-1], nil
+}
+
+// parseRegion first disregards anything before and including "codecommit" + double-colons/periods.
+// Ex: 	https://git-codecommit.us-west-2.amazonaws.com/v1/repos/aws-sample -> us-west-2.amazonaws.com/v1/repos/aws-sample
+//		codecommit::us-west-2://aws-sample -> us-west-2://aws-sample
+//		ssh://git-codecommit.us-west-2.amazonaws.com/v1/repos/aws-sample -> us-west-2.amazonaws.com/v1/repos/aws-sample
+// Then we take just the substring that precedes a period (HTTPS or SSH) or colon (git-remote-codecommit/federated),
+// check that it's an AWS region, and return it.
+func (o *initPipelineOpts) parseRegion(url string) (string, error) {
+	regexPattern := regexp.MustCompile(`(codecommit)(::|.)`)
```

Comment:
I wonder if it'd be a bit simpler to just do something dumb like this:
```go
if strings.HasPrefix(url, "https://") {
   // https parsing logic without regexes.
}
if strings.HasPrefix(url, "codecommit::") {
   
}
if strings.HasPrefix(url, "ssh://") {

}
return "", fmt.Errorf("unknown codecommit url format: %s", url)
}
```

Reply:
Okay. Maybe a switch?

Reply:
done!